### PR TITLE
Added : AWS dynamoDB post request  &  Improved : Items  posting structure

### PR DESCRIPTION
**Pull request by shivamsherje**
## Discription  : 
Added : AWS dynamoDB  post request to save the daily checkins feedbacks of users.
Improved: The method or the structure of posting the items in the dynamodb.
Added : Dependacies of dynamo DB aws-sdk
### Improved : code quality changes :
- Error handling
- Remove Unwanted code 
- Variable and function names.
- Added Map function instade of loop.
- Improve modal  to save the questions dyanamically instade of id.
- Improve modal to save the options  as number instade of its value.
- Improve functions.
- Update Sample.env
- Added id = user_id + timestamp

### CODE REVIEW

File: listeners/views/index.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const { sampleViewCallback } = require('./sample-view');
-const { openDailyCheckinModalCallback } = require('../actions/open-daily-checkin-action');
+const { postDailychechinFeedback } = require('./daily-checkins');
```
Comment (disha1202) [inline listeners/views/index.js]: Use camel case

Comment (disha1202) [inline listeners/views/index.js]:
```suggestion
const { postDailychechinFeedback } = require('./daily-checkin');
```

File: listeners/views/index.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
 
 module.exports.register = (app) => {
   app.view('sample_view_id', sampleViewCallback);
-  app.view('dailyCheckin', openDailyCheckinModalCallback);
+  app.view('dailyCheckin', postDailychechinFeedback);
```
Comment (disha1202) [inline listeners/views/index.js]: Improve variable name, should be postDailyCheckin

File: services/aws/index.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  region: 'ap-south-1',
+  endpoint: process.env.AWS_DYNAMO_DB_ENDPOINT_URL,
+  accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+};
+AWS.config.update(awsConfig);
+
+const docClient = new AWS.DynamoDB.DocumentClient();
+
+async function dailyCheckinsFeedback(params) {
```
Comment (disha1202) [inline services/aws/index.js]:
```suggestion
async function dailyCheckinFeedback(params) {
```

File: services/aws/index.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  const input = {
```
Comment (disha1202) [inline services/aws/index.js]: Why are we using an extra variable, move this to payload

File: services/aws/index.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    id: params.body.user.id + Math.random().toString(),
```
Comment (disha1202) [inline services/aws/index.js]: id should be combination of userid and timestamp

File: services/aws/index.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    User_id: params.body.user.id,
+    Timestamp: Math.floor(Date.now() / 1000),
+    Checkins: { },
```
Comment (disha1202) [inline services/aws/index.js]: Use camel case

File: services/aws/index.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  };
+
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < params.data.length; i++) {
```
Comment (disha1202) [inline services/aws/index.js]: Use array methods such as map, filter, reduce.

File: services/aws/index.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    const key = `Question_${i + 1}`;
```
Comment (disha1202) [inline services/aws/index.js]: Use an array of questions

File: services/aws/index.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+      Answer: params.data[i].answer,
+    };
+  }
+  const payload = {
+    TableName: 'DailyCheckin',
+    Item: input,
+  };
+  docClient.put(payload, (err) => {
+    if (err) {
+      console.log(`users::save::error - ${err}`);
```
Comment (disha1202) [inline services/aws/index.js]:
```suggestion
      console.error(`users::save::error - ${err}`);
```

File: services/aws/index.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
+const AWS = require('aws-sdk');
+require('dotenv').config();
+
+const awsConfig = {
+  region: 'ap-south-1',
+  endpoint: process.env.AWS_DYNAMO_DB_ENDPOINT_URL,
+  accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
```
Comment (disha1202) [inline services/aws/index.js:8]: Update this in .env.sample file as well

File: listeners/views/daily-checkins.js [hunk 11]
```diff
@@ -0,0 +1,19 @@
+const { dailyCheckinsFeedback } = require('../../services/aws/index');
+
+// eslint-disable-next-line consistent-return
+const postDailychechinFeedback = async ({ ack, body }) => {
+  await ack({
+    response_action: 'clear',
+  });
+  try {
+    const feedbackData = Object.values(body.view.state.values).map((value) => ({
+      question: Object.keys(value)[0],
```
Comment (disha1202) [inline listeners/views/daily-checkins.js:10]: The question should be the question, it is static select id right now.

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: services/aws/index.js (hunk 10 above)
Comment (shivamsherje) [reply services/aws/index.js:8]: done

File: services/aws/index.js (hunk 9 above)
Comment (shivamsherje) [reply services/aws/index.js]: done

File: services/aws/index.js (hunk 6 above)
Comment (shivamsherje) [reply services/aws/index.js]: done

File: services/aws/index.js (hunk 5 above)
Comment (shivamsherje) [reply services/aws/index.js]: done

File: services/aws/index.js (hunk 3 above)
Comment (shivamsherje) [reply services/aws/index.js]: done

File: listeners/views/index.js (hunk 2 above)
Comment (shivamsherje) [reply listeners/views/index.js]: done

File: listeners/views/index.js (hunk 1 above)
Comment (shivamsherje) [reply listeners/views/index.js]: done

Comment (shivamsherje) [reply listeners/views/index.js]: done

File: services/aws/index.js (hunk 4 above)
Comment (shivamsherje) [reply services/aws/index.js]: done

File: services/aws/index.js (hunk 7 above)
Comment (shivamsherje) [reply services/aws/index.js]: done

File: services/aws/index.js (hunk 8 above)
Comment (shivamsherje) [reply services/aws/index.js]: done

File: listeners/views/daily-checkins.js (hunk 11 above)
Comment (shivamsherje) [reply listeners/views/daily-checkins.js:10]: done

File: services/aws/index.js [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    question: item.question,
+    answer: item.answer,
+  }));
+
+  const payload = {
+    TableName: 'DailyCheckin',
+    Item: {
+      id: params.body.user.id + Math.floor(Date.now() / 1000),
+      user_id: params.body.user.id,
+      timestamp: Math.floor(Date.now() / 1000),
```
Comment (disha1202) [inline services/aws/index.js]: Create a global variable timestamp and use it everywhere

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: services/aws/index.js [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+  endpoint: process.env.AWS_DYNAMO_DB_ENDPOINT_URL,
+  accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+};
+AWS.config.update(awsConfig);
+
+const docClient = new AWS.DynamoDB.DocumentClient();
+
+async function dailyCheckinsFeedback(params) {
+  const checkin = params.data.map((item) => ({
```
Comment (disha1202) [inline services/aws/index.js]:
```suggestion
  const checkinFeedback = params.data.map((item) => ({
```

**Review by disha1202 — CHANGES_REQUESTED** [review]

**Review by disha1202 — APPROVED** [review]

**Pull request merged**